packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -45,25 +46,53 @@ def draw(self):
             self.figure.dpi = orig_dpi
             self._idle_scheduled = False
 
+    def print_png(self, filename_or_obj, *args,
+                  metadata=None, pil_kwargs=None,
+                  **kwargs):
+
+        from matplotlib import _png
+
+        if metadata is None:
+            metadata = {}
+        metadata = {
+            "Software":
+                f"matplotlib version{__version__}, http://matplotlib.org/",
+            **metadata,
+        }
+
+        canvas = self.get_element('canvas')
+        ctx = canvas.getContext("2d")
+        canvas_data = ctx.getImageData(0, 0, int(ctx.width),
+                                       int(ctx.height)).data
+        data = np.reshape(np.uint8(np.asarray(canvas_data)),
+                                  (int(ctx.height), int(ctx.width), -1))
```

Comment:
Let's move the above four lines out into its own function (maybe `get_pixel_data`).  When writing tests, this is all we'll need -- we won't need to encode the pixels as PNG -- so it would be good to have a place to just do that minimal amount of work.

Reply:
Yes, that makes sense, Should the new function `get_pixel_data` be inside the class `FigureCanvasHTMLCanvas` as well?

Also, on a sidenote, could you suggest a better name for the class `FigureCanvasHTMLCanvas`. The current name has `Canvas` 2 times which is a bit eerie

Reply:
Yes -- that function could go in the same class.

As for naming, I saw that, too, but I don't have any better suggestions.  The really confusing one is `FigureRendererHTMLCanvas`.  We could just replace `HTMLCanvas` with `HTML`...  But let's do that in a separate PR.

Reply:
Okay, thanks, working on these changes now.